source/components/hardware/hwxfsleep.c

```diff
@@ -180,13 +180,29 @@ AcpiHwSleepDispatch (
 #define ACPI_WAKE_PREP_FUNCTION_ID      1
 #define ACPI_WAKE_FUNCTION_ID           2
 
+/*
+ * Some compilers can handle designated initializers, which is needed
+ * under Linux kernel builds for structures that are entirely function
+ * pointers.
+ */
+#ifdef __GNUC__
+#define ACPI_SLEEP_FUNCTION_INIT(legacy, extended) \
+              { .legacy_function   = legacy,     \
+                .extended_function = extended }
```

Comment:
Is that possible to introduce a mechanism rather than a special case?
For example:
```
#define ACPI_DEFINE_FUNCITON(name, pointer) \
...
#define ACPI_SLEEP_FUNCTION_INIT(legacy, extended) \
{ \
    ACPI_DEFINE_FUNCITON(legacy_function, legacy), \
    ACPI_DEFINE_FUNCITON(extended_function, extended) \
}
```